src/main/java/use_case/delete_post/DeletePostInteractor.java

```diff
@@ -0,0 +1,67 @@
+package use_case.delete_post;
+
+import daos.DBUserDataAccessObject;
+import entity.Post;
+
+/**
+ * interactor for delete post
+ */
+public class DeletePostInteractor implements DeletePostInputBoundary {
+
+    private final DeletePostDataAccessInterface postDataAccessObject;
+    private final DeletePostOutputBoundary postPresenter;
+    private final DBUserDataAccessObject userRepo;
+
+    public DeletePostInteractor(DeletePostDataAccessInterface postDataAccessObject,
+                                DeletePostOutputBoundary postPresenter,
+                                DBUserDataAccessObject userRepo) {
+        this.postDataAccessObject = postDataAccessObject;
+        this.postPresenter = postPresenter;
+        this.userRepo = userRepo;
+    }
+
+    @Override
+    public void deletePost(DeletePostInputData deletePostInputData) {
+
+        Post post = postDataAccessObject.postExistsById(deletePostInputData.getPostId());
+
+        if (post == null) {
+            postPresenter.prepareFailView("Post not found.");
+            throw new DeletePostFailedException("The post with ID " + deletePostInputData.getPostId() + " does not exist.");
+        }
+
+        boolean userCanDelete = this.canDelete(deletePostInputData.getUserId(), post);
```

Comment:
by this logic, userCanDelete will always be true - instead u should getCurrentUser() from the user dao (you'll have to pass it in the constructor) and use that to check instead